sqlmesh/dbt/context.py

```diff
@@ -144,13 +145,20 @@ def models(self) -> t.Dict[str, ModelConfig]:
     def models(self, models: t.Dict[str, ModelConfig]) -> None:
         self._models = {}
         self._refs = {}
+        self._models_by_fqn = {}
         self.add_models(models)
 
     def add_models(self, models: t.Dict[str, ModelConfig]) -> None:
         self._refs = {}
         self._models.update(models)
         self._jinja_environment = None
 
+    @property
+    def models_by_fqn(self) -> t.Dict[str, ModelConfig]:
```

Comment:
Looks like this should just be `@cached_property` which returns a set of FQNs. No need for this to be a dict